ironfish/src/storage/levelup/transaction.ts

```diff
@@ -27,23 +29,35 @@ export class LevelupTransaction implements IDatabaseTransaction {
   waitingResolve: (() => void) | null = null
   id = 0
 
+  public lockAcquisition: Meter
+  public lockContention: Meter
+  public lockWaitTime: number
   static id = 0
 
   constructor(db: LevelupDatabase) {
     this.db = db
     this.id = ++LevelupTransaction.id
     this.batch = new LevelupBatch(db)
+    this.lockAcquisition = new Meter()
+    this.lockAcquisition.start()
+    this.lockContention = new Meter()
+    this.lockContention.start()
+    this.lockWaitTime = 0
   }
 
   get size(): number {
     return this.batch.queue.length
   }
 
   async acquireLock(): Promise<void> {
+    this.lockAcquisition.add(1)
+    const start = BenchUtils.start()
+
     if (this.unlock) {
       return
     }
 
+    this.lockContention.add(1)
```

Comment:
Similar here this is number of lock contentions per second. What would we use this number for?

Reply:
I am thinking to use them in this way: number of lock contention / number of lock acquisition, this will be the contention ratio. If the ratio is high then we know we have contention issue.